Parsers/ASimAuditEvent/Parsers/vimAuditEventMicrosoftSecurityEvents.yaml

```diff
@@ -0,0 +1,288 @@
+        4700, "Enable Scheduled Task", "Enable", "", "Scheduled Task", "Success",
+        4701, "Disable Scheduled Task ", "Disable", "", "Scheduled Task", "Success",
+        4702, "Update Scheduled Task", "Set", "", "Scheduled Task", "Success",
+        4929, "Remove Active Directory Replica Source Naming Context", "Delete", "", "Other", "Success",
+        5025, "Stop Firewall Service", "Disable", "Firewall Service", "Service", "Success",
+        5027, "Retrieve the Security Policy From The Local Storage", "Read", "Firewall Service", "Service", "Failure",
+        5028, "Parse the new Security Policy", "Set", "Firewall Service", "Service", "Failure",
+        5029, "Initialize the Firewall Driver", "Initialize", "Firewall Service", "Service", "Failure",
+        5030, "Start the Firewall Service", "Start", "Firewall Service", "Service", "Failure",
+        5034, "Stop Firewall Driver", "Stop", "Firewall Driver", "Driver", "Failure",
+        5035, "Start Firewall Driver", "Start", "Firewall Driver", "Driver", "Failure",
+        5037, "Terminating Firewall Driver", "Terminate", "Firewall Driver", "Driver", "Failure",
+        7035, "Start Control Sent", "Execute", "Service", "Service", "Success",
+        7036, "Enter Stop State", "Stop", "Service", "Service", "Success",
+        7040, "Changed Service Settings", "Set", "Service", "Service", "Success",
+        7045, "Install Service", "Install", "Service", "Service", "Success",
+        2009, "Load Group Policy", "Other", "Service", "Service", "Failure",
+        5136, "Modified Directory Services Object", "Set", "", "Directory Service Object", "Success"
+    ];
+        let FilteredEventIds = toscalar(EventIDLookup 
+            | where (array_length(eventtype_in) == 0 or EventType in (eventtype_in))
+                and (array_length(operation_has_any) == 0 or Operation has_any (operation_has_any))
+                and (eventresult == '*' or EventResult == eventresult)
+                and EventID != 1102 // Exclude this EventID, we have separate section for including EventID 1102
+            | summarize make_set(EventID)
+            );
+        let ParsedEvents =
+            union
+                (
+                    // SecurityEvents
+                    SecurityEvent
+                    | where not(disabled)
+                    | where (isnull(starttime) or TimeGenerated >= starttime) 
+                        and (isnull(endtime) or TimeGenerated <= endtime)
+                        and EventID in(FilteredEventIds)
+                    | where (array_length(srcipaddr_has_any_prefix) == 0)
+                        and (array_length(actorusername_has_any) == 0 or EventData has_any (actorusername_has_any))
+                        and (array_length(newvalue_has_any) == 0 or EventData has_any (newvalue_has_any))         
+                    | project EventID, EventData, _ResourceId, TimeGenerated, Computer, Type, _ItemId
```

Comment:
done automated testing. Following are the results, can you please check?
FAIL: tests_main_func (__main__.FilteringTest.tests_main_func) (<subtest>)
----------------------------------------------------------------------
AssertionError: 0 == 0 : Parameter: actorusername_has_any - Got no results at all after filtering. Filtered by value: @'SECCXP\billb'

======================================================================
FAIL: tests_main_func (__main__.FilteringTest.tests_main_func) (<subtest>)
----------------------------------------------------------------------
AssertionError: 0 == 0 : Parameter: actorusername_has_any - Got no results at all after filtering. Filtered by value: @'SECCXP\billb',@'NT AUTHORITY\SYSTEM'

======================================================================
FAIL: tests_main_func (__main__.FilteringTest.tests_main_func) (<subtest>)
----------------------------------------------------------------------
AssertionError: 1 == 1 : Only one value exists for parameter: eventresult - validations for this parameter are partial

----------------------------------------------------------------------
Ran 1 test in 142.026s

FAILED (failures=3)
::error::Tests failed for Parsers/ASimAuditEvent/Parsers/vimAuditEventMicrosoftSecurityEvents.yaml

Reply:
![Uploading image.png…]()

Reply:
can't see the picture

Reply:
testing done ASC portal